### PR TITLE
Version comparing: sort 1.x versions last 

### PR DESCRIPTION
**Pull request by stsewd**
Fixes https://github.com/readthedocs/readthedocs.org/issues/10046

### CODE REVIEW

File: readthedocs/projects/version_handling.py [hunk 1]
```diff
@@ -41,7 +37,9 @@ def parse_version_failsafe(version_string):
     except InvalidVersion:
         # Handle the special case of 1.x, 2.x or 1.0.x, 1.1.x
         if final_form and '.x' in final_form:
-            return parse_version_failsafe(final_form.replace('.x', '.0'))
+            # Replace the .x with .999999 so it's sorted last.
+            final_form = final_form.replace(".x", ".999999")
```
[the remark this answers is not in the document]
Comment (stsewd) [reply readthedocs/projects/version_handling.py:41]: I think it's fine to just replace all occurrences (code is simpler), or do you have an example where this is a problem?

[the remark this answers is not in the document]
Comment (stsewd) [reply readthedocs/projects/version_handling.py:41]: We also support for suffixes, like `1.x.rc`, so we can't just always rely on the last position.

[the remark this answers is not in the document]
Comment (benjaoming) [reply readthedocs/projects/version_handling.py:41]: Ah okay, that's a good point.

Well, maybe your suggestion is a good one, and it's better to proceed with the simple implementation and then polish it later :+1:

**Review by benjaoming — APPROVED** [review]
LGTM :rocket: 

Thanks for the quick exchange on how to do the string replacement, I think you're right :+1:

**Pull request merged**